EIPS/eip-7727.md

```diff
@@ -0,0 +1,228 @@
+
+An example would of the `transactionList` would be:
+
+```go
+[
+	DELEGATED_TX_TYPE.payload, 
+	BUNDLE_TX_TYPE.payload, 
+	DELEGATED_TX_TYPE.payload
+]
+```
+
+The `signatureYParity`, `signatureR`, `signatureS` elements of the `BUNDLE_TX_TYPE` represent a secp256k1 signature over:
+
+```go
+keccak256(0x06 || rlp([bundleSigner, blockNumber, chainId, nonce, gasPrice, transactionList]))
+```
+
+We’ll refer to address resolved by this signature the bundle transaction’s signer. 
+
+### `BUNDLE_SIGNER` Opcode
+
+The `BUNDLE_SIGNER` is a new opcode, added in `FORK_BLOCK`, identified by `BUNDLE_SIGNER_OPCODE_NUMBER` and requires zero stack arguments.
+
+When the transaction type is `DELEGATED_TX_TYPE`, this opcode pushes the `bundleSigner` from the transaction payload onto the stack as an address. If the transaction is of a different type, it returns the zero address.
+
+The gas cost for this opcode is `BUNDLE_SIGNER_OPCODE_COST`.
+
+### Transaction Validity Rules
+
+For a `DELEGATED_TX_TYPE` to be valid, the following must be true:
+
+- The transaction must be included in a `BUNDLE_TX_TYPE`'s `transactionList` to be valid.
+- The transaction’s specified `bundleSigner` must be equal to the address who signed over the `BUNDLE_TX_TYPE` that included the transaction in a `transactionList`.
+- The payload variable `blockNumber`, if not zero, must be the block number that the transaction is executed in.
+
+For a `BUNDLE_TX_TYPE` to be valid, the following MUST be true:
+
+- All transactions in the `transactionList` must specify the signer of the `BUNDLE_TX_TYPE` as the `bundleSigner`.
+- The transaction must be included in a `BUNDLE_TX_TYPE`'s `transactionList` if the `bundleSigner` payload variable is not the zero address.
+- The payload variable `blockNumber`, if not zero, must be the block number that the transaction is executed in.
+
+### Gas Costs
+
+The `BUNDLE_TX_TYPE` has a new gas cost formula. For the `BUNDLE_TX_TYPE` to be valid, it must have enough gas to cover:
+
+1. The `BUNDLE_TX_TYPE` transaction base cost of `BUNDLE_BASE_GAS_COST`.
+2. All bytes of the `transactionList` elements as if their bytes were part of a typical transaction’s `CALL_DATA`.
+
+The final gas cost of the `BUNDLE_TX_TYPE` depends on if the `transactionList`'s elements are valid transactions or not. The following logic is used on the `transactionList` elements:
+
+1. If an element is valid and can begin execution, the `BUNDLE_TX_TYPE` signer is not charged for that element’s bytes.
+2. If the element is not valid and cannot being execution, the `BUNDLE_TX_TYPE` is charged as if that element’s bytes were just `CALL_DATA`.
+
+At the start of processing, the `BUNDLE_TX_TYPE`'s signer must be charged as if all inner transactions were invalid and refunded the gas for the valid transactions as each valid transaction finishes execution.
```

Comment:
Only include the actual gas formula in the specification section. The reasoning behind it should live in the rationale section. Helps to keep the specification as small and readable as possible.

Reply:
Do you mean I should have a code block instead of a verbal description? The words here are describing different parts of the gas costing logic and aren't justifications for it.

Reply:
A code block would definitely be fine, but it isn't necessary. I think I just misinterpreted the way you wrote this, honestly.

Instead of writing it as "it must have enough gas to cover...", I'd recommend wording it like:

> The gas cost of BUNDLE_TX_TYPE is calculated as follows `BUNDLE_BASE_GAS_COST` plus `FOO` times `BAR`, where `FOO` is zero if [...] and is seven otherwise.

Introducing the formula with "must have enough gas to cover" just doesn't sound precise enough.

Reply:
I updated it! Let me know if you think it still needs work, happy to refactor again.